R/helpers_knitr_CohortSize.R

```diff
@@ -0,0 +1,229 @@
+#' @importFrom knitr knit_print
+NULL
+
+# Integration with knitr ----
+#'
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' We provide additional utility functions to allow human-friendly rendition of
+#' crmPack objects in Markdown and Quarto files
+#'
+#' @return a character string that represents the object in markdown.
+#' @name knit_print
+NULL
+
+# CohortSize ----
+
+#' Render a CohortSizeConst Object
+#'
+#' @description `r lifecycle::badge("experimental")`
+#' @param x (`CohortSize`)\cr The object to knit_print.
+#' @param asis (`flag`)\cr Should the return value be wrapped in a call to `knitr::asis_output`?
+#' @param label (`character`)\cr The word used to label the participants.  See Usage Notes below.
+#' @param ... Not used at present
+#'
+#' @section Usage Notes:
+#' `label` describes the trial's participants.
+#'
+#' It should be a character vector of length 1 or 2.  If of length 2, the first
+#' element describes a `cohort_size` of 1 and the second describes all other
+#' `cohort_size`s.  If of length 1, the character `s` is appended to the value
+#' when `cohort_size` is not 1.
+#' @return The markdown representation of the object, as a character string
+#' @seealso [`knit_print`] for more details.
+#'
+#' @export
+#' @rdname knit_print
+knit_print.CohortSizeConst <- function(x, ..., asis = TRUE, label = c("participant", "participants")) {
+  assert_character(label, min.len = 1, max.len = 2, any.missing = FALSE)
+  assert_flag(asis)
+
+  if (length(label) == 1) {
+    label[2] <- paste0(label[1], "s")
+  }
+  rv <- paste0("A constant size of ", x@size, " ", label[ifelse(x@size == 1, 1, 2)], ".")
+  if (asis) {
+    rv <- knitr::asis_output(rv)
+  }
+  rv
+}
+registerS3method("knit_print", "CohortSizeConst", knit_print.CohortSizeConst)
+
+#' Render a CohortSizeRange Object
+#'
+#' @description `r lifecycle::badge("experimental")`
+#' @inherit knit_print.CohortSizeConst return
+#'
+#' @export
+#' @rdname knit_print
+knit_print.CohortSizeRange <- function(x, ..., asis = TRUE) {
+  assert_flag(asis)
+
+  param <- list(...)
+  if (!("col.names" %in% names(param))) {
+    param[["col.names"]] <- c("Lower", "Upper", "Cohort size")
+  }
+  if (!("caption" %in% names(param))) {
+    param[["caption"]] <- "Defined by the dose to be used in the next cohort"
+  }
+  x <- x %>% tidy()
```

Comment:
```suggestion
  x <- tidy(x)
```
generally let's avoid pipe in package code, and especially for such usage normal function call is shorter